From 968d3549f84cd1f529328c4cd8d968c189d6d646 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston <jeremyhu@apple.com>
Date: Sun, 15 Aug 2010 20:07:28 -0700
Subject: [PATCH] rootless: Remove ROOTLESS_WORKAROUND

This was already removed for XWin (20701522be803fe47e921fcf059dadf64c7f287d)
with no reported side effects.  XQuartz seems to be behaving ok without it as
well.  While this possibly brings back bug #1168, we don't have any reproduction
steps for that issue, and if it crops up again, we should fix it a real way
rather than this hokey workaround which doesn't even work for COMPOSITE.

Signed-off-by: Jeremy Huddleston <jeremyhu@apple.com>
---
 configure.ac          |    2 +-
 miext/damage/damage.c |    6 +-----
 2 files changed, 2 insertions(+), 6 deletions(-)

diff --git xorg-server-1.9.0/configure.ac xorg-server-1.9.0/configure.ac
index 9884fa7..95f7a76 100644
--- xorg-server-1.9.0/configure.ac
+++ xorg-server-1.9.0/configure.ac
@@ -1926,7 +1926,7 @@ if test "x$XQUARTZ" = xyes; then
 
 	AC_CHECK_LIB([Xplugin],[xp_init],[:])
 
-	CFLAGS="${CFLAGS} -DROOTLESS_WORKAROUND -DROOTLESS_SAFEALPHA -DNO_ALLOCA"
+	CFLAGS="${CFLAGS} -DROOTLESS_SAFEALPHA -DNO_ALLOCA"
 
 	PKG_CHECK_MODULES(XPBPROXY, $APPLEWMPROTO $LIBAPPLEWM xfixes x11)
 
diff --git xorg-server-1.9.0/miext/damage/damage.c xorg-server-1.9.0/miext/damage/damage.c
index 1cf0513..e0e96f2 100644
--- xorg-server-1.9.0/miext/damage/damage.c
+++ xorg-server-1.9.0/miext/damage/damage.c
@@ -89,11 +89,7 @@ getDrawableDamageRef (DrawablePtr pDrawable)
 	ScreenPtr   pScreen = pDrawable->pScreen;
 
 	pPixmap = 0;
-	if (pScreen->GetWindowPixmap
-#ifdef ROOTLESS_WORKAROUND
-	    && ((WindowPtr)pDrawable)->viewable
-#endif
-	    )
+	if (pScreen->GetWindowPixmap)
 	    pPixmap = (*pScreen->GetWindowPixmap) ((WindowPtr)pDrawable);
 
 	if (!pPixmap)
-- 
1.5.6.6

